src/main/java/org/archive/extract/WATExtractorOutput.java

```diff
@@ -68,13 +75,45 @@ public void output(Resource resource) throws IOException {
 	}
 
 	private void writeWARCInfo(OutputStream recOut, MetaData md) throws IOException {
-		String filename = JSONUtils.extractSingle(md, "Container.Filename");
-		if(filename == null) {
-			throw new IOException("No Container.Filename...");
+		// filename is given in the command line
+		String filename = outputFile;
+		if (filename == null || filename.length() == 0) {
+			// if no filename by command line, we construct a default filename base on container filename
+			filename = JSONUtils.extractSingle(md, "Container.Filename");
+			if (filename == null) {
+				throw new IOException("No Container.Filename...");
+			}
+			if (filename.endsWith(".warc") || filename.endsWith(".warc.gz")) {
+				filename = filename.replaceFirst("\\.warc$", ".warc.wat.gz");
+				filename = filename.replaceFirst("\\.warc\\.gz$", ".warc.wat.gz");
+			} else if (filename.endsWith(".arc") || filename.endsWith(".arc.gz")) {
+				filename = filename.replaceFirst("\\.arc$", ".arc.wat.gz");
+				filename = filename.replaceFirst("\\.arc\\.gz$", ".arc.wat.gz");
+			}
 		}
+		// removing path from filename
+		File tmpFile = new File(filename);
+		filename = tmpFile.getName();
 		HttpHeaders headers = new HttpHeaders();
-		headers.add("Software-Info", IAUtils.COMMONS_VERSION);
-		headers.addDateHeader("Extracted-Date", new Date());
+		headers.add("software", IAUtils.COMMONS_VERSION);
+		headers.addDateHeader("extractedDate", new Date());
+		
+		// add ip, hostname, format, etc.
+		headers.add("ip", InetAddress.getLocalHost().getHostAddress());
```

Comment:
@scheylord I ran the WAT extractor code on a different server than usual today and it threw a java.net.UnknownHostException on the getLocalHost() here. I was able to avoid getting the exception by modifying the server's /etc/hosts file, but we may want to catch that exception similar to the [Heritrix code] (https://github.com/internetarchive/heritrix3/blob/master/modules/src/main/java/org/archive/modules/writer/WARCWriterProcessor.java#L774) again. Beyond that, the change you made to getCanonicalHostName() worked for me.